### PR TITLE
chore: bump common API version to 3.3.1

### PR DESCRIPTION
**Pull request by typotter**
Picks up latest from common package.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -31,7 +31,7 @@ repositories {
 
 dependencies {
   // Re-export classes and interfaces that will be used upstream
-  api 'cloud.eppo:sdk-common-jvm:3.0.2'
+  api 'cloud.eppo:sdk-common-jvm:3.2.0-SNAPSHOT'
```
Comment (typotter) [inline build.gradle:34]: Using snapshot until release is cut/pushed

File: src/test/java/com/eppo/sdk/EppoClientTest.java [hunk 2]
```diff
@@ -276,7 +276,7 @@ private void uninitClient() {
   private void initBuggyClient() {
     try {
       EppoClient eppoClient = initClient(DUMMY_FLAG_API_KEY);
-      Field configurationStoreField = BaseEppoClient.class.getDeclaredField("requestor");
+      Field configurationStoreField = BaseEppoClient.class.getDeclaredField("configurationStore");
```
Comment (typotter) [inline src/test/java/com/eppo/sdk/EppoClientTest.java:279]: base implementation changed to access config only through the store.

**Review by felipecsl — APPROVED** [review]

File: src/main/java/com/eppo/sdk/EppoClient.java [hunk 3]
```diff
@@ -125,6 +139,12 @@ public Builder host(String host) {
       return this;
     }
 
+    /** Sets the initial configuration for the client. */
+    public Builder initialConfiguration(CompletableFuture<Configuration> initialConfiguration) {
```
Comment (aarsilv) [inline src/main/java/com/eppo/sdk/EppoClient.java:143]: 🔥

**Review by aarsilv — APPROVED** [review]
Thanks for sneaking this one in!

**Comment by aarsilv** [comment]
Obviously don't merge & release until common is published and tests pass here 😛

**Pull request merged**